compiler/ast/Python.asdl

```diff
@@ -60,7 +60,7 @@ module Python
          | UnaryOp(unaryop op, expr operand)
          | Lambda(arguments args, expr body)
          | IfExp(expr test, expr body, expr orelse)
-         | Dict(expr* keys, expr* values)
+         | Dict(expr?* keys, expr* values)
```

Comment:
CPython doesn't have `?` in keys anymore.

https://github.com/python/cpython/blob/main/Parser/Python.asdl#L64

Reply:
@youknowone True, but `dict.keys` does contain `None`.

Reply:
https://github.com/python/cpython/blob/main/Parser/Python.asdl#L64 used to look like `| Dict(expr?* keys, expr* values)?`

Reply:
It looks like this line hasn't changed for 11 years: https://github.com/python/cpython/blame/3325f054e33b318aa56b74472f76a56b8afc0510/Parser/Python.asdl#L64

Reply:
https://github.com/python/cpython/blob/3325f054e33b318aa56b74472f76a56b8afc0510/Grammar/python.gram#L904

```
double_starred_kvpair[KeyValuePair*]:
    | '**' a=bitwise_or { _PyPegen_key_value_pair(p, NULL, a) }
                                                     ^^^^
    | kvpair
```

Reply:
Posted https://github.com/python/cpython/issues/101211

Reply:
Oh, then `expr?* keys` must have been our local patch. I didn't know that. Thank you.

Reply:
@youknowone Can we move forward with `expr?* keys`?

Reply:
I don't think `expr?*` is the right thing here. Not only is it _probably_ not valid ASDL syntax (see [here](https://github.com/eliben/asdl_parser/blob/master/docs/ASDL%20Zephyr%20-%20Wang.pdf), the modifiers are in brackets as opposed to braces, though this might not really be an issue), but it is also kinda ambiguous, is it an Optional list or a list of optionals?

Can't we get the same behavior by modifying `asdl_rs` and keeping `Python.asdl` the same as CPythons?

Reply:
> Not only is it probably not valid ASDL syntax (see [here](https://github.com/eliben/asdl_parser/blob/master/docs/ASDL%20Zephyr%20-%20Wang.pdf), the modifiers are in brackets as opposed to braces, though this might not really be an issue), but it is also kinda ambiguous, is it an Optional list or a list of optional?

Agreed!

> Can't we get the same behavior by modifying asdl_rs and keeping Python.asdl the same as CPythons?

We can. I'm happy to revert the last commit.

@youknowone What do you think?

Reply:
If we could do, I prefer to keep Python.asdl as same as CPython.

Reply:
Yes, updated the code.